dotcom-rendering/src/web/lib/useApi.tsx

```diff
@@ -27,12 +27,14 @@ interface ApiResponse<T> {
  * returns { loading, error, data }
  * @param {String} url - The url to fetch
  * @param {SWRConfiguration} options - The SWR config object - https://swr.vercel.app/docs/options
+ * @param {RequestInit} init - The fetch init object - https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#supplying_request_options
  * */
 export const useApi = <T,>(
 	url: string,
 	options?: SWRConfiguration,
+	init?: RequestInit,
 ): ApiResponse<T> => {
-	const { data, error } = useSWR(url, fetcher, options);
+	const { data, error } = useSWR(url, fetcher(init), options);
```

Comment:
Maybe using `Data` instead of `T` this type is more explicit, and `extends unkown` is better than the odd dangling comma to prevent TSX from tripping up?

You can then also pass these types to the useSWR command, like so:

```suggestion
export const useApi = <Data extends unknown>(
	url: string,
	options?: SWRConfiguration,
	init?: RequestInit,
): ApiResponse<Data> => {
	const { data, error } = useSWR<Data, Error>(url, fetcher(init), options);
```

Reply:
Personally, I'm not especially triggered by the dangly comma. If the linting tooling needs this there then I'm very happy to let it do its thing. But maybe I'm missing something, do you think it could cause an issue?

Reply:
That said, I'm very happy to make this change if it solves the problem!

But do you mind if we branch this off from this PR as this work is blocking another PR which is itself blocking the larger dev build work which I'm keen to progress!

Reply:
The dangling comma [seems to be what people expect](https://github.com/prettier/prettier/issues/6302). I hold my mind’s breath, but I probably should just let it go… ;)

Reply:
I've created the PR for this change here: https://github.com/guardian/dotcom-rendering/pull/3840 - happy wither way, your call!